Modules/VersionManager/VersionManager.lua

```diff
@@ -0,0 +1,153 @@
+        if lag ~= 0 or uncommitted ~= 0 then return end
+        if not self._initialized then
+            LOG:Message("Classic Loot Manager %s initialization complete.", ColorCodeText(CLM.CORE:GetVersionString(), "00cc00"))
+            C_Timer.After(math.random(1, 5), function()
+                self:AnnounceVersion()
+            end)
+            self._initialized = true
+        end
+    end)
+
+    MODULES.ConfigManager:RegisterUniversalExecutor("ver", "Version", self)
+end
+
+function VersionManager:AnnounceVersion()
+    LOG:Trace("VersionManager:AnnounceVersion()")
+    local version = CLM.CORE:GetVersion()
+    local message = VersionCommStructure:New(
+        CONSTANTS.VERSIONNING_COMM.TYPE.ANNOUNCE_VERSION,
+        VersionCommAnnounceVersion:New(version.major, version.minor, version.patch, version.changeset))
+    Comms:Send(VERSION_COMM_PREFIX, message, CONSTANTS.COMMS.DISTRIBUTION.GUILD)
+end
+
+function VersionManager:RequestVersion()
+    LOG:Trace("VersionManager:RequestVersion()")
+    local message = VersionCommStructure:New(CONSTANTS.VERSIONNING_COMM.TYPE.REQUEST_VERSION, {})
+    Comms:Send(VERSION_COMM_PREFIX, message, CONSTANTS.COMMS.DISTRIBUTION.GUILD)
+end
+
+function VersionManager:DirectSendVersion(target)
+    LOG:Trace("VersionManager:AnnounceVersion()")
+    local version = CLM.CORE:GetVersion()
+    local message = VersionCommStructure:New(
+        CONSTANTS.VERSIONNING_COMM.TYPE.ANNOUNCE_VERSION,
+        VersionCommAnnounceVersion:New(version.major, version.minor, version.patch, version.changeset))
+    Comms:Send(VERSION_COMM_PREFIX, message, CONSTANTS.COMMS.DISTRIBUTION.WHISPER, target)
+end
+
+function VersionManager:HandleIncomingMessage(message, distribution, sender)
+    LOG:Trace("VersionManager:HandleIncomingMessage()")
+    local mtype = message:Type() or 0
+    if self.handlers[mtype] then
+        self[self.handlers[mtype]](self, message:Data(), sender)
+    end
+end
+
+function VersionManager:OutOfDate(version, disable)
+    LOG:Trace("VersionManager:OutOfDate()")
+    local currentTime = GetServerTime()
+    if disable then
+        if currentTime - self._lastDisplayedMessageD > 300 then
+            LOG:Message("|cffcc0000Your Classic Loot Manager is significantly out of date.|r AddOn communication has been disabled. Version %s is available. Please update as soon as possible.", ColorCodeText(stringifyVersion(version), "00cc00"))
+            self._lastDisplayedMessageD = currentTime
+        end
+        LedgerManager:Cutoff()
+        Comms:Disable()
+    else
+        if currentTime - self._lastDisplayedMessage > 300 then
+            LOG:Message("New version %s of Classic Loot Manager is available. For best experience please update the AddOn.", ColorCodeText(stringifyVersion(version), "00cc00"))
+            self._lastDisplayedMessage = currentTime
+        end
+    end
+end
+
+function VersionManager:HandleAnnounceVersion(data, sender)
```

Comment:
Make this a local function and use an actual function reference in your map of handlers (instead of a string)